### ISSUE TITLE
[Testing] Raycast capabilities

### ISSUE THREAD

**Issue by Axel1092** [carla-simulator/carla#3508]
New functions have been added: `world.cast_ray(init, end)`, `world.project_point(pos, dir, dist)` and `world.ground_projection(pos, dist)`. These return `LabelledPoint` which is a position in space and a semantic tag. 
`cast_ray` return intersections between the line connecting init and end and the geometry of the scene.
`project_point` returns only the first intersection of a particular point and a specified direction.
`ground_projection`is a specialized version of `project_point`

**Comment by joel-mb** [comment]
@Axel1092 works great!

**Comment by zizo1111** [comment]
Hi @Axel1092 @joel-mb @germanros1987 ,
do these methods follow the exact shape of the geometries?
I have tried out the methods and found that, when casting many points around an object the resulting shape is a simplified representation of how it looks like. I have noticed also that a point cloud produced by the Lidar provides more details about the geometries.
Is this really the case, or is it a problem on my side.
Thanks in advance.
